### PR TITLE
Pass additional security tokens for authentication before EJB invocation.

### PR DESCRIPTION
**Pull request by darranl**
An additional quick start to demonstrate passing an additional security token used for authentication of the remote user before the call is accepted for the EJB invocation.

### CODE REVIEW

**Comment by sgilda** [comment]
@darranl : This is great! I did have a problem though.
1. I configured the server. The CLI commands worked great.
2. I added the user.
3. I built and deployed the quickstart. This is the log:

09:47:07,970 INFO  [org.jboss.as.repository](management-handler-thread - 10) JBAS014900: Content added at location /home/sgilda/tools/jboss-as-7.2.0.Alpha1-SNAPSHOT/standalone/data/content/55/ccc035f0d79be6d59ccb2b01231291dcb58408/content
09:47:07,972 INFO  [org.jboss.as.server.deployment](MSC service thread 1-6) JBAS015876: Starting deployment of "jboss-as-ejb-security-plus.jar" (runtime-name: "jboss-as-ejb-security-plus.jar")
09:47:07,988 WARN  [org.jboss.as.dependency.private](MSC service thread 1-4) JBAS018567: Deployment "deployment.jboss-as-ejb-security-plus.jar" is using a private module ("org.jboss.as.domain-management:main") which may be changed or removed in future versions without notice.
09:47:07,989 WARN  [org.jboss.as.dependency.private](MSC service thread 1-4) JBAS018567: Deployment "deployment.jboss-as-ejb-security-plus.jar" is using a private module ("org.jboss.as.domain-management:main") which may be changed or removed in future versions without notice.
09:47:07,989 WARN  [org.jboss.as.dependency.private](MSC service thread 1-4) JBAS018567: Deployment "deployment.jboss-as-ejb-security-plus.jar" is using a private module ("org.jboss.as.controller:main") which may be changed or removed in future versions without notice.
09:47:07,989 WARN  [org.jboss.as.dependency.private](MSC service thread 1-4) JBAS018567: Deployment "deployment.jboss-as-ejb-security-plus.jar" is using a private module ("org.jboss.as.controller:main") which may be changed or removed in future versions without notice.
09:47:07,993 INFO  [org.jboss.as.ejb3.deployment.processors.EjbJndiBindingsDeploymentUnitProcessor](MSC service thread 1-8) JNDI bindings for session bean named SecuredEJB in deployment unit deployment "jboss-as-ejb-security-plus.jar" are as follows:

```
java:global/jboss-as-ejb-security-plus/SecuredEJB!org.jboss.as.quickstarts.ejb_security_plus.SecuredEJBRemote
java:app/jboss-as-ejb-security-plus/SecuredEJB!org.jboss.as.quickstarts.ejb_security_plus.SecuredEJBRemote
java:module/SecuredEJB!org.jboss.as.quickstarts.ejb_security_plus.SecuredEJBRemote
java:jboss/exported/jboss-as-ejb-security-plus/SecuredEJB!org.jboss.as.quickstarts.ejb_security_plus.SecuredEJBRemote
java:global/jboss-as-ejb-security-plus/SecuredEJB
java:app/jboss-as-ejb-security-plus/SecuredEJB
java:module/SecuredEJB
```

09:47:08,121 INFO  [org.jboss.as.server](management-handler-thread - 10) JBAS018559: Deployed "jboss-as-ejb-security-plus.jar" (runtime-name : "jboss-as-ejb-security-plus.jar")
1. When I run `mvn exec:exec`, I get the following error and no message in the server log:

sgilda@sgilda-T510 ~/GitRepos/quickstart-jdf/ejb-security-plus (jdf-pull-482) $ mvn exec:exec
[INFO] Scanning for projects...
[INFO]  
[INFO] ------------------------------------------------------------------------
[INFO] Building JBoss AS Quickstarts: EJB Security Plus 7.1.2-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:1.2.1:exec (default-cli) @ jboss-as-ejb-security-plus ---

---

Exception in thread "main" java.lang.IllegalStateException: EJBCLIENT000025: No EJB receiver available for handling [appName:, moduleName:jboss-as-ejb-security-plus, distinctName:] combination for invocation context org.jboss.ejb.client.EJBClientInvocationContext@10f102d3
    at org.jboss.ejb.client.EJBClientContext.requireEJBReceiver(EJBClientContext.java:693)
    at org.jboss.ejb.client.ReceiverInterceptor.handleInvocation(ReceiverInterceptor.java:116)
    at org.jboss.ejb.client.EJBClientInvocationContext.sendRequest(EJBClientInvocationContext.java:183)
    at org.jboss.ejb.client.EJBInvocationHandler.doInvoke(EJBInvocationHandler.java:177)
    at org.jboss.ejb.client.EJBInvocationHandler.doInvoke(EJBInvocationHandler.java:161)
    at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:124)
    at $Proxy0.getPrincipalInformation(Unknown Source)
    at org.jboss.as.quickstarts.ejb_security_plus.RemoteClient.main(RemoteClient.java:43)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.517s
[INFO] Finished at: Mon Mar 25 09:48:57 EDT 2013
[INFO] Final Memory: 12M/212M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.2.1:exec (default-cli) on project jboss-as-ejb-security-plus: Command execution failed. Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 

I'll go back through it and see if I missed something in the setup.

**Comment by sgilda** [comment]
I think I found the problem. 
- The password specified for 'UserOne`in the README file is`PasswordOne1!`. 
- The password used in RemoteClient.java is `password1!`.

Will test that theory next.

**Comment by sgilda** [comment]
That was it! In the README file, change the password for UserOne to `password1!` and it works great!

**Comment by sgilda** [comment]
I'm good with this.
@pmuir : Just waiting for your review. :-)

File: ejb-security-plus/README.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+
+### RemoteClient
+
+Finally there is the `RemoteClient` stand-alone client. The client demonstrates how a Principal with a custom credential can be set using the `SecurityContextAssociation` and how this can subsequently be used by a `CallbackHandler` for the username/password authentication required for the SASL authentication and subsequently how the additional token can be picked up by a client side interceptor and passed to the server with the invocation.
+
+
+Note on EJB client interceptors
+-----------------------
+
+JBoss Enterprise Application Platform 6.1 and JBoss AS 7.2 allow client side interceptors for EJB invocations. Such interceptors are expected to implement the `org.jboss.ejb.client.EJBClientInterceptor` interface. User applications can then plug in such interceptors in the 'EJBClientContext' either programatically or through the ServiceLoader mechanism.
```
Comment (pmuir) [inline ejb-security-plus/README.md:41]: JBoss AS 7.2 doesn't really exist, so just go with JBoss Enterprise Application Platform.

**Comment by pmuir** [comment]
Merged

**Pull request closed without merging**